### ISSUE TITLE
Current summary and icon sometimes returning `none`

### ISSUE THREAD

**Issue by SiriosDev** [Pirate-Weather/pirateweather#281]
### Describe the bug

In recent days I have been experiencing repeated appearances of the `exceptional` state. Checking on merrysky it does not turn out to be anything particularly out of the ordinary

### Expected behavior

See an analogue to the conditions shown by other PW-based 

### Actual behavior

e.g. exactly now it is shown exceptional when the condition is trivially partly cloudy
(`eccezionale` mean `exceptional`)
![image](https://github.com/user-attachments/assets/84521587-01d7-4d38-bd58-0d2543bd0ab5)
![image](https://github.com/user-attachments/assets/b25bc22a-c593-4fe2-bbba-cb5287933f09)

### Home Assistant version

2024.7.4

### Integration version

1.5.2

### Other details

Unfortunately, the `exceptional` condition is poorly documented but browsing the core code [these](https://github.com/home-assistant/core/blob/537d7728a726816220be0ca7c7f68ecea7d835d1/homeassistant/components/nws/const.py#L34) seem to be the conditions that fall under the term `exceptional`

> ```py
> #...
> CONDITION_CLASSES: dict[str, list[str]] = {
>     ATTR_CONDITION_EXCEPTIONAL: [
>         "Tornado",
>         "Hurricane conditions",
>         "Tropical storm conditions",
>         "Dust",
>         "Smoke",
>         "Haze",
>         "Hot",
>         "Cold",
>     ],
> #...
> ```

### Troubleshooting steps

- [X] I have updated my Home Assistant installation to the latest version.
- [X] I have updated the Pirate Weather Integration to the latest version.
- [X] I have gone through the [documentation](https://pirateweather.net/en/latest/ha/) before opening this issue.
- [X] I have searched this repository and [API Repository](https://github.com/alexander0042/pirate-weather-ha) to see if the issue has already been reported.
- [X] I have restarted my Home Assistant installation.
- [X] I have queried the API in my browser to confirm the issue is not with the API.
- [X] I have written an informative title.

**Comment by cloneofghosts** [comment]
I've moved this issue over to the API repository since it's an issue with the API and not the HA integration. I'll leave this one open for now until we can confirm that V2.1 of the API will fix this issue as it could be a different issue entirely.

**Comment by alexander0042** [comment]
Thanks for flagging this- it's been tricky to try to get fixed, so every report helps! Do you mind letting me know where you're located, since that'll help me find the offending process flow

**Comment by SiriosDev** [comment]
at the moment nothing strange, however the position is around this 39.51181975109912, 15.945201813230057

**Comment by alexander0042** [comment]
Looked into this today and I'm 90% sure I know the issue here! What seems to be happening is an issue with the interpolation for the minutely data, essentially when the probability/ amount of precipitation increases significantly from one 3-hourly output timestep to the next, it can get into a situation where the the amount is enough to trigger a rain icon but there's no categorical percentage yet. 

Long story short, there's two parts to the fix. The first is adding a check to see if this is occurring and fixing it (easy), and secondly is switching the interpolation approach to be more consistent. Testing both the changes on the dev endpoint at the moment, and will be rolled out in 2.1!

**Comment by alexander0042** [comment]
Ok, finally got to the bottom of this, and it turns out that it's (mostly) not a bug in my code for once! GEFS will sometimes (not often) return no precipitation type despite modelling precipitation. If you want to validate it, look at latitude=24.0, longitude=261.5, on 2024-08-15 21:00:00 using the 2024-08-14 T06Z forecast, where there's a mean accumulation of 0.032 mm but no categorical precipitation.

That being said, the API response should be able to handle this, and the obvious answer is to set precipitation to zero, since it always seems to occur with very small amounts. probabilities. I'll update the code to handle that, but it's good to know I'm not crazy here
